src/Morphir/Spark/AST.elm

```diff
@@ -67,12 +67,55 @@ These are the supported Transformations:
           - A List of (Name, Expression) which represent a alias for a column expression and a column expression
             like `col(name)` within spark.
           - A target ObjectExpression from which to select.
+  - **Aggregate**
+      - Represents a `df.groupBy(...).agg(...)` transformation.
+      - The arguments are:
+          - A String, which represents the column to group aggregations by.
+          - A list of named expressions, which represent the aggregation methods being used on this dataset.
+          - The ObjectExpression to aggregate the results of.
 
 -}
 type ObjectExpression
     = From ObjectName
     | Filter Expression ObjectExpression
     | Select NamedExpressions ObjectExpression
+    | Aggregate String NamedExpressions ObjectExpression
+
+
+{-| An AggregationCall represents a call to Morphir.SDK.Aggregate.aggregate
+
+Its values are:
+
+  - **Group Key**
+      - i.e. the name of the column results are grouped by.
+  - **Returned Group Key**
+      - Optionally, the name to give the group column in the results.
+  - **Aggregate Values**
+      - i.e. the expressions that define columns in the output of aggregate.
+  - **Source Value**
+      - i.e. the Value that is being aggregated
+
+-}
+type AggregationCall
+    = AggregationCall Name (Maybe Name) (List AggregateValue) TypedValue
+
+
+{-| An AggregateValue represents a single aggregation used within the overall Aggregation call
+
+Its values are:
+
+  - **Field Name**
+      - i.e. The column name in the output.
+  - **Filter**
+      - A Maybe Value of a filter expression, if it exists.
+  - **Aggregation Name**
+      - The Name of the aggregation function used (e.g. "sumOf")
+  - **Aggregation Arg**
+      - Any argument passed to the aggregation (e.g. ".ageOfItem" in "sumOf .AgeOfItem")
+
+-}
+type AggregateValue
+    = AggregateValue Name (Maybe TypedValue) FQName (Maybe TypedValue)
```

Comment:
If I understand correctly these (`AggregationCall` and `AggregateValue`) are structures that represent a specific API pattern in the Morphir IR that are easier to process than the raw expressions. If that's the case, I think they should be moved into `Morphir.IR.SDK.List` since they should be aligned with the List API and don't seem to be specific to the Spark backend.

Reply:
You understand correctly.

I am not sure how `Morphir.IR.SDK.List` fits into the rest of morphir, and the code in there looks pretty intimidating, so I'd appreciate some guidance on how to do this.

Reply:
What you see there is code that composes types directly using the raw IR constructs and also some native functions that make the interpreter work. But you don't need to understand the code that's in there. You can simply move this type definition together with the `constructAggregationCall` function into that module. At least from what I can see these don't depend on anything in the Spark AST.

Reply:
Thanks! Will do.

Reply:
Sorry, I made a mistake when I told you to move this into `Morphir.IR.SDK.List`. What I should have said is to move it to `Morphir.IR.SDK.Aggregate`. Can you please move it again, sorry.